syfertext/tokenizer.py

```diff
@@ -175,18 +217,411 @@ def __call__(self, text: Union[String, str]):
+                    start_pos=start_pos + pos,
+                    end_pos=end_pos + pos,
+                    space_after=False,  #  For this token space_after will be updated explicitly in end.
+                    is_space=False,
+                )
+
+                # Append the token to the infix_list
+                infix_tokens_metas.append(token_meta)
+
+        # We have already proccesed thr full substring so updating
+        # pos just to have similar structure to other _get_affix_meta.
+        pos = len(substring) + 1
+
+        # There is no remaining substring
+        substring = ""
+
+        return infix_tokens_metas, substring, pos
+
+    def _get_exception_token_metas(self, substring: str, pos: int) -> Tuple[List[TokenMeta], str]:
+        """Make a list of TokenMeta objects of exceptions found in `substring`.
+
+        Args:
+            substring: The substring to tokenize.
+            pos: The pointer to location of start of substring in text.
+
+        Returns:
+            exception_token_metas : the list of exceptions TokenMeta 
+                objects.
+            substring: The updated substring after processing the exceptions.
+
+        """
+        exception_token_metas = []
+
+        for e in self.exceptions[substring]:
+            ORTH = e["ORTH"]
+            end_pos = pos + len(ORTH) - 1
+
+            # Create the TokenMeta object
+            token_meta = TokenMeta(
+                hash_key=self.vocab.store[ORTH],
+                start_pos=pos,
+                end_pos=end_pos,
+                space_after=False,  # for the last token space_after will be updated explicitly in end.
+                is_space=False,
+            )
+
+            # Append the token to the  exception tokens list
+            exception_token_metas.append(token_meta)
+
+            # update start_pos for next orth
+            pos = end_pos + 1
+
+        # There is no remaining substring.
+        substring = ""
+
+        return exception_token_metas, substring
+
+    def infix_matches(self, substring: str) -> List[Match]:
+        """Find internal split points of the string, such as hyphens.
+        
+        Args:
+            substring : The string to segment.
+
+        Returns:
+            A list of `re.MatchObject` objects that have `.start()`
+                and `.end()` methods, denoting the placement of internal 
+                segment separators, e.g. hyphens.
+        """
+        if self.infix_finditer is None:
+            return []
+        return list(self.infix_finditer(substring))
+
+    def find_prefix(self, substring: str) -> int:
+        """Find the length of a prefix that should be segmented from the
+        string, or None if no prefix rules match.
+
+        Args:
+            substring: The string to segment.
+            
+        Returns:
+            The length of the prefix if present, otherwise 0.
+        """
+        if self.prefix_search is None:
```

Comment:
empty line after the docstring + some commenting. 

Please in all your code, add empty lines that separate different code blocks, such as after `return 0` below. And another empty line before `return (match.end() -.....`

Reply:
done